code/__DEFINES/callbacks.dm

```diff
@@ -1,6 +1,23 @@
 #define GLOBAL_PROC "some_magic_bullshit"
 /// A shorthand for the callback datum, [documented here](datum/callback.html)
 #define CALLBACK new /datum/callback
-#define INVOKE_ASYNC world.ImmediateInvokeAsync
+
+///Per the DM reference, spawn(-1) will execute the spawned code immediately until a block is met.
+#define MAKE_SPAWN_ACT_LIKE_WAITFOR -1
+///Create a codeblock that will not block the callstack if a block is met.
+#define ASYNC spawn(MAKE_SPAWN_ACT_LIKE_WAITFOR)
+
+#define INVOKE_ASYNC(proc_owner, proc_path, proc_arguments...) \
+	if ((proc_owner) == GLOBAL_PROC) { \
+		ASYNC { \
+			call(proc_path)(##proc_arguments); \
+		}; \
+	} \
+	else { \
+		ASYNC { \
+			call(proc_owner, proc_path)(##proc_arguments); \
+		}; \
+	} \
```

Comment:
Why are you escaping this newline? Seems unnecessary - if someone eventually puts something below it'll eat it

Reply:
error